packages/protocol/contracts/shared/tokenvault/ERC20Vault.sol

```diff
@@ -296,18 +316,27 @@ contract ERC20Vault is BaseVault {
         // Don't send the tokens back to `from` because `from` is on the source chain.
         checkToAddress(to);
 
-        // Transfer the ETH and the tokens to the `to` address
-        address token = _transferTokens(ctoken, to, amount);
+        address tokenRecipient = to;
+
+        // If the bridging intent has been solved, the solver becomes the token recipient
+        address solver = solverConditionToSolver[solverCondition];
+        if (solver != address(0)) {
+            tokenRecipient = solver;
```

Comment:
how about we also delete the mapping item in side the if-statement:

`delete solverConditionToSolver[solverCondition]`

Reply:
Agreed. It shall give a refund and it is no longer useful hereforth.